pyvista/plotting/render_window_interactor.py

```diff
@@ -1528,75 +1553,176 @@ def close(self):
+        self.OnLeftButtonUp()
+        parent = self._parent()
+        if len(parent._plotter.renderers) > 1:  # type: ignore[union-attr]
+            for renderer in parent._plotter.renderers:  # type: ignore[union-attr]
+                renderer.SetInteractive(True)
+
+    def add_observer(self, event, callback):
+        """Keep track of observers.
+
+        Parameters
+        ----------
+        event : str
+            VTK event string
+        callback : callable
+            Function to call during callback
+
+        """
+        self._observers.append(self.AddObserver(event, callback))
+
+    def remove_observers(self):  # numpydoc ignore=SS06
+        """Remove all observers added through
+        :func:`~pyvista.plotting.render_window_interactor.CaptureInteractorStyle.add_observer`.
+        """  # noqa : D205
+        for obs in self._observers:
+            self.RemoveObserver(obs)
 
-            def _press(self, *args):
-                # Figure out which renderer has the event and disable the
-                # others
-                super().OnLeftButtonDown()
-                parent = self._parent()
-                if len(parent._plotter.renderers) > 1:  # type: ignore[union-attr]
-                    click_pos = parent.get_event_position()  # type: ignore[union-attr]
-                    for renderer in parent._plotter.renderers:  # type: ignore[union-attr]
-                        interact = renderer.IsInViewport(*click_pos)
-                        renderer.SetInteractive(interact)
-
-            def _release(self, *args):
-                super().OnLeftButtonUp()
-                parent = self._parent()
-                if len(parent._plotter.renderers) > 1:  # type: ignore[union-attr]
-                    for renderer in parent._plotter.renderers:  # type: ignore[union-attr]
-                        renderer.SetInteractive(True)
-
-            def add_observer(self, event, callback):
-                self._observers.append(self.AddObserver(event, callback))
-
-            def remove_observers(self):
-                for obs in self._observers:
-                    self.RemoveObserver(obs)
-
-        return CustomStyle
-
-    # cache classes
-    if klass not in _CLASSES:
-        _CLASSES[klass] = _make_class(klass)
-    return _CLASSES[klass]
+
+# All interactor styles here inherit from `CaptureInteractorStyle`, which
+# inherits from `DisableVtkSnakeCase`, so don't duplicate again.
+class InteractorStyleImage(CaptureInteractorStyle, _vtk.vtkInteractorStyleImage):
+    """Image interactor style.
+
+    See Also
+    --------
+    pyvista.plotting.render_window_interactor.RenderWindowInteractor.enable_image_style
+
+    """
+
+
+class InteractorStyleJoystickActor(CaptureInteractorStyle, _vtk.vtkInteractorStyleJoystickActor):
+    """Joystick actor interactor style.
+
+    See Also
+    --------
+    pyvista.plotting.render_window_interactor.RenderWindowInteractor.enable_joystick_actor_style
```

Comment:
When I search for this in the current docs https://dev.pyvista.org/search.html?q=enable_joystick, I see results for
[pyvista.Plotter.enable_joystick_actor_style](https://dev.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.enable_joystick_actor_style.html#pyvista.Plotter.enable_joystick_actor_style) and [pyvista.RenderWindowInteractor.enable_joystick_actor_style](https://dev.pyvista.org/api/plotting/_autosummary/pyvista.RenderWindowInteractor.enable_joystick_actor_style.html#pyvista.RenderWindowInteractor.enable_joystick_actor_style), so I think the links need to be one of these, e.g.:
```suggestion
    pyvista.RenderWindowInteractor.enable_joystick_actor_style
```

or make it shorter to simply show it as "enable_joystick_actor_style" if desired:

```suggestion
    :class:`~pyvista.RenderWindowInteractor.enable_joystick_actor_style`
```

Reply:
Whoops, I think :meth: is needed instead of :class: ...

Reply:
I switched to using :meth:. I also got rid of `~` now that the resolving path is shorter.